src/host/main.cpp

```diff
@@ -215,15 +215,32 @@ int main(int argc, char** argv)
           "On start, ledger directory should not exist ({})",
           config.ledger.directory));
       }
-      // Count members with public encryption key as only these members will be
-      // handed a recovery share.
-      // Note that it is acceptable to start a network without any member having
-      // a recovery share. The service will check that at least one recovery
-      // member is added before the service can be opened.
+
+      for (auto const& m : config.command.start.members)
+      {
+        if (
+          !m.encryption_public_key_file.has_value() &&
+          m.recovery_owner.has_value())
+        {
+          throw std::logic_error(fmt::format(
+            "No public encryption key has been specified but recovery owner "
+            "value has been set for a member"));
+        }
+      }
+
+      // Count members with public encryption key who are not recovery
+      // owners as only these members will be handed a recovery share
+      // that accrues towards the recovery threshold.
+      // Note that it is acceptable to start a network without any member
+      // having a recovery share. The service will check that at least one
```

Comment:
I believe we support opening deliberately unrecoverable services, and although I am not aware of current use cases, they have come up as potential use cases in the past, so I think we want to leave that open as a possibility.

Reply:
@achamayou not sure which lines you wanted me to change here. I added (a) a check to ensure that if recovery_owner has a value then enc_pub_key must also have a value else throw and (b) count the member_with_pubk_count while skipping the recovery_owner members.
(a) is like a configuration issue while (b) is only ensuring the correctness of the existing check that the count of recovery members and supplied or calculated default recovery threshold values are sane else the logic throws below.

Reply:
@achamayou  "The service will check that at least one..." this check that there must be atleast 1 recovery member (aka participant) existed before this PR and continues to work today. So as of now you cannot open a service that has 0 recovery participants. I preserved the check in main.cpp using members_with_pubk_count and there are checks in set_recovery_threshold, remove_member and open_service that continue to work as before. Having owners has not changed the checks around recovery threshold and recovery members (participants).